### DISCUSSION TITLE
load all necessary records in searchNext request

### DISCUSSION THREAD

**Discussion by wnleao** [sevdokimov/log-viewer#48] [Ideas]
Could you take a look at https://github.com/sevdokimov/log-viewer/pull/49 and let me know what you think? 

I was working testing some things in the frontend and I thought that loading all necessary records and sending them back in the same search response would lead to a better flow when searching for terms. This way we don't see the flickering (with loading progress) in the screen anymore and we get the next occurrence index in the search response as well.

**Comment by sevdokimov** [comment]
Thanks. Will review today/tomorrow.

**Reply by wnleao** [reply]
Okay! Let me know if you have any questions. I haven't found much time in my schedule this week, but I'll get back to working/testing on Saturday.

**Reply by sevdokimov** [reply]
Good idea. But keep in mind that loading records may take a lot of time. It takes time when the log is huge and user specified filters that hide most lines. In this case, we should return the result as soon as possible, loading lines after the result may be in background. 
Ideally, log-viewer should try to load the next lines during 100ms, if loading takes more than 100ms it should return found line without waiting for the next lines. Let's implement it. Or I can modify your PR if you don't have time.

**Reply by wnleao** [reply]
Okay! I'll take a look into that.

**Reply by wnleao** [reply]
Hi! I've updated the commit, check if that's what you have in mind. Let me know what you think.

**Reply by sevdokimov** [reply]
Hi!
Thank you for the PR. Search looks much better when no flickering.
I merged your PR into [search_next_load_all](https://github.com/sevdokimov/log-viewer/tree/search_next_load_all) branch and slightly modified it. Please, look.

**Reply by wnleao** [reply]
Hey! Sure, I'm gonna take a look.

**Reply by wnleao** [reply]
Great changes, thank you! The only question that I have is related to the evaluation of loadNext after a search is performed. If loadNext is false, but there is some data available, shouldn't you compute and return the corresponding "foundIdx", either 0 or data.length-1? In the current scenario -1 will be returned.

**Reply by sevdokimov** [reply]
yes, you are right. Fixed.

**Reply by wnleao** [reply]
nice! I've created the PR.

**Comment by sevdokimov** [comment]
Done. The fix is merged to "master" branch.
